### PR TITLE
Merging code to add the no API flag.

### PR DESCRIPTION
**Pull request by barnesalex**
- Added a -n flag to allow running on deploy ramdisk.
- Updated Readme for the entire repo

### CODE REVIEW

File: bin/g5k-checks [hunk 1]
```diff
@@ -38,7 +38,7 @@ optparse = OptionParser.new(USAGE) do |opts|
   end
 
   $hashconf["mode"] = "oar_checks"
-  opts.on( '-m', '--mode MODE', "Specify the functional mode (oar_checks, api, jenkins) [default: 'oar_checks']") do |mode|
+  opts.on( '-m', '--mode MODE', "Specify the functional mode (oar_checks, api, jenkins, no_api) [default: 'oar_checks']") do |mode|
```
Comment (diurnalist) [inline bin/g5k-checks]: I don't think we need this.

File: lib/g5kchecks/utils/node.rb [hunk 2]
```diff
@@ -39,7 +39,10 @@ def initialize(conf)
       @ohai_description = nil
       @api_description = nil
       @max_retries = 2
-      get_openstack_vendor_data
+      if @conf["noapi"] != true
```
Comment (diurnalist) [inline lib/g5kchecks/utils/node.rb]: Could do `unless @conf["noapi"]`, it is equivalent in Ruby

File: testing_update_rpm.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#! /bin/bash
+
+./scripts/build_rpm.sh
+yum -y remove rubygem-g5k-checks
```
Comment (diurnalist) [inline testing_update_rpm.sh]: Why is this part needed?

**Review by diurnalist — APPROVED** [review]
Looks good! Left a few comments/questions but nothing major. :+1:

File: bin/g5k-checks (hunk 1 above)
Comment (barnesalex) [reply bin/g5k-checks]: Removed, good catch! This was left overs from when I was trying to make a new mode.

File: lib/g5kchecks/utils/node.rb (hunk 2 above)
Comment (barnesalex) [reply lib/g5kchecks/utils/node.rb]: Implemented and tested.

File: testing_update_rpm.sh (hunk 3 above)
Comment (barnesalex) [reply testing_update_rpm.sh]: I used this script to build and install the gem on my test node.  I don't think it is necessary to bring into the master branch.

**Comment by barnesalex** [comment]
Made all changes and removed some extraneous files.
